src/components/GridContainer/GridContainer.tsx

```diff
@@ -0,0 +1,129 @@
+  rowGap?: GapOptions;
+}
+
+const GridContainer = ({
+  alignItems = "stretch",
+  alignContent = "stretch",
+  children,
+  columnGap = "none",
+  display = "grid",
+  gap = "unset",
+  gridAutoColumns = "",
+  gridAutoFlow = "row",
+  gridAutoRows = "",
+  gridTemplateAreas = "",
+  gridTemplateColumns = "",
+  gridTemplateRows = "",
+  gridTemplate = "",
+  isResponsive = true,
+  justifyContent = "stretch",
+  justifyItems = "stretch",
+  rowGap = "none",
+  ...props
+}: GridContainerProps) => {
+  return (
+    <Wrapper
+      $alignItems={alignItems}
+      $alignContent={alignContent}
+      $columnGap={columnGap}
+      $display={display}
+      $gap={gap}
+      $gridAutoColumns={gridAutoColumns}
+      $gridAutoFlow={gridAutoFlow}
+      $gridAutoRows={gridAutoRows}
+      $gridTemplateAreas={gridTemplateAreas}
+      $gridTemplateColumns={gridTemplateColumns}
+      $gridTemplateRows={gridTemplateRows}
+      $gridTemplate={gridTemplate}
+      $isResponsive={isResponsive}
+      $justifyContent={justifyContent}
+      $justifyItems={justifyItems}
+      $rowGap={rowGap}
+      data-testid="grid-container"
+      {...props}
+    >
+      {children}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div<{
+  $alignContent: ContentOptions;
+  $alignItems: ItemsOptions;
+  $columnGap: GapOptions;
+  $display: DisplayOptions;
+  $gap: GapOptions;
+  $gridAutoColumns: string;
+  $gridAutoFlow: FlowOptions;
+  $gridAutoRows: string;
+  $gridTemplateAreas: string;
+  $gridTemplateColumns: string;
+  $gridTemplateRows: string;
+  $gridTemplate: string;
+  $isResponsive: boolean;
+  $justifyContent: ContentOptions;
+  $justifyItems: ItemsOptions;
+  $rowGap: GapOptions;
+}>`
+  align-items: ${({ $alignItems = "stretch" }) => $alignItems};
+  align-content: ${({ $alignContent = "stretch" }) => $alignContent};
+  column-gap: ${({ theme, $columnGap }) => theme.click.gridContainer.gap[$columnGap]};
+  display: ${({ $display }) => $display};
+  gap: ${({ theme, $gap }) => theme.click.gridContainer.gap[$gap]};
+  grid-auto-columns: ${({ $gridAutoColumns }) => $gridAutoColumns || "auto"};
+  grid-auto-flow: ${({ $gridAutoFlow }) => $gridAutoFlow || "auto"};
+  grid-auto-rows: ${({ $gridAutoRows }) => $gridAutoRows || "auto"};
+  grid-template-area: ${({ $gridTemplateAreas }) => $gridTemplateAreas || "auto"};
+  grid-template-columns: ${({ $gridTemplateColumns }) => $gridTemplateColumns || "auto"};
+  grid-template-rows: ${({ $gridTemplateRows }) => $gridTemplateRows || "auto"};
+  grid-template: ${({ $gridTemplate }) => $gridTemplate || "auto"};
+  justify-content: ${({ $justifyContent = "stretch" }) => $justifyContent};
+  justify-items: ${({ $justifyItems = "stretch" }) => $justifyItems};
+  row-gap: ${({ theme, $rowGap }) => theme.click.gridContainer.gap[$rowGap]};
+
+  background: rgba(30, 30, 30, 0.8);
+
+  @media (max-width: ${({ theme }) => theme.breakpoint.sizes.md}) {
+    grid-template-columns: ${({ $isResponsive = true }) =>
```

Comment:
We have grid columns but not rows here or area any reason for excluding them here?

Reply:
Just because v1 of mobile is similar to the flex approach, 1 column, full width. What would you suggest we do with rows here?

Reply:
I think there might be a problem in the UI with rows and columns defined together
Lets iterate on it later